### PR TITLE
fix: enable overlays in _enable_craft_parts_features()

### PR DESCRIPTION
**Pull request by tigarmo**
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

**Review by mr-cal — APPROVED** [review]

**Comment by sergiusens** [comment]
```
sergiusens@cuivienen:~/Source/canonical/rockcraft$ snap refresh --amend --classic rockcraft --channel edge/pr-602
rockcraft (edge/pr-602) 1.4.0.post11+g64fe731 from Canonical✓ refreshed
sergiusens@cuivienen:~/Source/canonical/rockcraft$ mkdir foo
sergiusens@cuivienen:~/Source/canonical/rockcraft$ cd foo/
sergiusens@cuivienen:~/Source/canonical/rockcraft/foo$ rockcraft init
deleting current features configuration
sergiusens@cuivienen:~/Source/canonical/rockcraft/foo$ ls
rockcraft.yaml

**Comment by sergiusens** [comment]
nevermind! I was using my venv by accident :/

**Review by sergiusens — APPROVED** [review]

File: tests/conftest.py [hunk 1]
```diff
@@ -248,3 +248,20 @@ def _inner(**kwargs):
         )
 
     return _inner
+
+
+@pytest.fixture(autouse=True)
+def reset_features():
```
Comment (sergiusens) [inline tests/conftest.py:254]: there is no need for this to be an independent fixture, is there?

Could this not be cleaner

```python
@pytest.fixture()
def enable_overlay_feature(reset_features):
    """Enable the overlay feature."""
    from craft_parts import Features

    Features.reset()
    Features(enable_overlay=True)
    yield
    Features.reset()
```

?

Comment (tigarmo) [reply tests/conftest.py:254]: > there is no need for this to be an independent fixture, is there?

There is, actually. If this feature is not used in *all* tests, then two consecutive tests that call `Application.run()` will try to enable the overlay feature twice and fail. It's a side-effect of the `Features` singleton.
I add it as a parameter explicitly in the `enable_overlay_feature()` fixture to ensure correct order: first `reset_features` is setup, then `enable_overlay_feature`, then the test runs, then teardown in the reverse order

**Pull request merged**